hazelcast/src/main/java/com/hazelcast/ringbuffer/Ringbuffer.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.ringbuffer;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.core.IFunction;
+import com.hazelcast.spi.annotation.Beta;
+
+import java.util.Collection;
+
+/**
+ * A Ringbuffer is a data-structure where the content is stored in a ring like structure. A ringbuffer has a capacity so it
+ * won't grow beyond that capacity and endanger the stability of the system. If that capacity is exceeded, than the oldest
+ * item in the ringbuffer is overwritten.
```

Comment:
As far as I understand, this behaviour depends on `OverflowPolicy`, right? If capacity is exceeded, depending on the overflow-policy, oldest item is overwritten or operation fails...